pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -810,6 +810,33 @@ public SimpleHttpResponse uploadSegment(URI uri, String segmentName, File segmen
     return uploadSegment(uri, segmentName, segmentFile, null, parameters, DEFAULT_SOCKET_TIMEOUT_MS);
   }
 
+  /**
```

Comment:
I had to add a new method to FileUploadDownloadClient because the existing methods don't accept enableParallelPushProtection as a parameter. This new method is used by my changes in the integration test.

Reply:
Agreed that we could use existing API. But IMO having this method provides a convenient wrapper around the existing API. I envision that any new parameters added to Restlet.UploadSegment() API will also be added here.